adapter/src/dbt/adapters/fal_experimental/adapter.py

```diff
@@ -29,29 +31,46 @@ def run_with_adapter(code: str, adapter: BaseAdapter) -> Any:
         write_df=prepare_for_adapter(adapter, write_df_to_relation),
     )
 
-def _isolated_runner(code: str, config: RuntimeConfig) -> Any:
+def _isolated_runner(code: str, config: RuntimeConfig, manifest: Manifest, macro_manifest: MacroManifest) -> Any:
     # This function can be run in an entirely separate
     # process or an environment, so we need to reconstruct
     # the DB adapter solely from the config.
-    adapter = reconstruct_adapter(config)
+    adapter = reconstruct_adapter(config, manifest, macro_manifest)
     return run_with_adapter(code, adapter)
 
 def run_in_environment_with_adapter(
     environment: BaseEnvironment,
     code: str,
     config: RuntimeConfig,
+    manifest: Manifest,
+    macro_manifest: MacroManifest
 ) -> AdapterResponse:
+    from isolate.backends.remote import IsolateServer
     """Run the 'main' function inside the given code on the
     specified environment.
 
     The environment_name must be defined inside fal_project.yml file
     in your project's root directory."""
+    deps = [i for i in get_default_pip_dependencies() if i.startswith('dbt-')]
+    if type(environment) == IsolateServer:
+        if environment.target_environment_kind == 'conda':
+            environment.target_environment_config['packages'].extend(deps)
```

Comment:
I am still not sure about this, considering `deps` are pip packages and not conda (e.g.`x==y` vs `x=y`, plus there might be packages that doesn't exist on the conda forge).

But it also means we can't solve it for variadic dbt dependencies now, since we can't create multiple environments. So maybe, now is the time to actually implement inheritance among environments? (In any case, for this PR, I'd recommend just raising an error when the target is conda on remote; since there is no easy fix).